### PR TITLE
chore: update to yarn 4 & Node 18

### PR DESCRIPTION
**Pull request by jstarpl**
## About Me

This pull request is posted on behalf of the NRK.

## Type of Contribution

This is a: 
Code improvement 

## Current Behavior

Package Manager depends on a no-longer maintained `pkg`, which also means it's stuck on Node 16, which is now EOL.

## New Behavior

`pkg` is replaced with `@yao-pkg/pkg`, which also means that an upgrade to Node 18 (possibly to 20 too) is now possible. Yarn is upgraded to yarn 3. The testing matrix is expanded to include Node 18 (obviously) and Node 20, so that we are ready to upgrade as soon as possible.

## Testing Instructions

* See if the automatic builds still work.

## Status

- [x] PR is ready to be reviewed.
- [x] The functionality has been tested by the author.
- [x] Relevant unit tests has been added / updated.
- [ ] Relevant documentation (code comments, [system documentation](https://nrkno.github.io/sofie-core/)) has been added / updated.

### CODE REVIEW

File: .github/workflows/build-windows.yaml [hunk 1]
```diff
@@ -18,18 +18,20 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Enable Corepack
```
Comment (Julusian) [inline .github/workflows/build-windows.yaml:21]: These `Enable Corepack` should be after `Use Node.js`.

As it is currently done, it is doing the `corepack enable` using the corepack from the default nodejs (which looks to be 18.19.1 currently), then it will be replacing corepack and nodejs.

File: .github/workflows/lint-and-test.yaml [hunk 2]
```diff
@@ -68,7 +77,7 @@ jobs:
 
       - name: Prepare Environment
         run: |
-          yarn config set network-timeout 100000 -g
+          corepack enable
```
Comment (Julusian) [inline .github/workflows/lint-and-test.yaml:80]: having to do a `corepack enable` twice here is weird. hopefully this will not be necessary after my earlier comment

File: apps/single-app/app/expectedPackages.json
Comment (Julusian) [inline apps/single-app/app/expectedPackages.json:1]: was this change intentional?

**Review by Julusian — APPROVED** [review]
Is it worthwhile keeping support for node16, or should it switch entirely to 18+?

Only supporting 18+ would allow for using yarn 4 instead of 3 and it is something we will want to do 'soon' anyway, so its really just a question of whether to do it all at once now, or in two stages.

File: apps/single-app/app/expectedPackages.json
Comment (jstarpl) [reply apps/single-app/app/expectedPackages.json:1]: No, it wasn't. Thanks for catching this.

File: .github/workflows/lint-and-test.yaml (hunk 2 above)
Comment (jstarpl) [reply .github/workflows/lint-and-test.yaml]: I just started off with someone's tutorial saying that the `corepack enable` should be before `setup-node` action, but I guess the tutorial was just wrong.

**Comment by jstarpl** [comment]
> Is it worthwhile keeping support for node16, or should it switch entirely to 18+?
> Only supporting 18+ would allow for using yarn 4 instead of 3 and it is something we will want to do 'soon' anyway, so its really just a question of whether to do it all at once now, or in two stages.

I'm entirely for dropping 18 support entirely, if @nytamin doesn't object. I'll upgrade the Dockerfiles too.

**Review by nytamin — APPROVED** [review]
